js/coinex.js

```diff
@@ -3092,6 +3093,89 @@ module.exports = class coinex extends Exchange {
         };
     }
 
+    parseTransfer (transfer, currency = undefined) {
+        //
+        // fetchTransfers
+        //
+        //     {
+        //         "amount": "10",
+        //         "asset": "USDT",
+        //         "transfer_type": "transfer_out", // from swap to spot
+        //         "created_at": 1651633422
+        //     },
+        //
+        const timestamp = this.safeTimestamp (transfer, 'created_at');
+        const transferType = this.safeString (transfer, 'transfer_type');
+        let fromAccount = undefined;
+        let toAccount = undefined;
+        if (transferType === 'transfer_out') {
+            fromAccount = 'swap';
+            toAccount = 'spot';
+        } else if (transferType === 'transfer_in') {
+            fromAccount = 'spot';
+            toAccount = 'swap';
+        }
+        if (currency === undefined) {
+            currency = this.safeString (transfer, 'asset');
```

Comment:
We use safeCurrencyCode instead of safeString for the purpose of converting from exchange-specific currency ids to unified currency codes:

```
const currencyId = this.safeString (transfer, 'asset');
const code = this.safeCurrencyCode (currencyId, currency);
// ...
return {
    'currency': code,
};
```